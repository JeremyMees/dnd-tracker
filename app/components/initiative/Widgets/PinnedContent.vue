<script setup lang="ts">
defineEmits<{ update: [Open5eItem[]] }>()
defineProps<{ value: Open5eItem[] }>()
</script>

<template>
  <Card
    color="secondary"
  >
    <UiAccordion
      v-if="value.length"
      type="single"
      collapsible
    >
      <UiAccordionItem
        v-for="item in value"
        :key="item.slug"
        :value="item.slug"
        class="last:border-b-0"
      >
        <UiAccordionTrigger class="hover:no-underline">
          {{ item.name }}
        </UiAccordionTrigger>
        <UiAccordionContent>
          <ContentCardContent
            :content="item"
            :is-open="true"
            :hide-open-button="() => true"
            :type="item.type"
          />
          <div class="flex justify-end pt-4">
            <button
              :aria-label="$t(`components.infoCard.remove`)"
              class="btn-primary flex items-center gap-2"
              @click="$emit('update', value.filter(i => i.slug !== item.slug))"
            >
              <Icon
                name="tabler:pinned-off"
                class="size-4 min-w-4"
                aria-hidden="true"
              />
              {{ $t('components.infoCard.remove') }}
            </button>
          </div>
        </UiAccordionContent>
      </UiAccordionItem>
    </UiAccordion>
    <p
      v-else
      class="text-muted-foreground"
    >
      {{ $t('pages.encounter.pinnedContent.empty') }}
    </p>
  </Card>
</template>
