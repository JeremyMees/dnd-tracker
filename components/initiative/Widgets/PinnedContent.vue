<script setup lang="ts">
const emit = defineEmits<{ update: [Open5eItem[]] }>()

const props = defineProps<{ value: Open5eItem[] }>()
</script>

<template>
  <Card
    color="secondary"
  >
    <UiAccordion
      type="single"
      collapsible
    >
      <UiAccordionItem
        v-for="item in props.value"
        :key="item.slug"
        :value="item.slug"
        class="last:border-b-0"
      >
        <UiAccordionTrigger class="hover:no-underline">
          {{ item.name }}
        </UiAccordionTrigger>
        <UiAccordionContent>
          <ContentCardContent
            :content="item"
            :is-open="true"
            :hide-open-button="() => true"
            :type="item.type"
          />
          <div class="flex justify-end pt-4">
            <button
              :aria-label="$t(`components.infoCard.remove`)"
              class="btn-primary flex items-center gap-2"
              @click="emit('update', value.filter(i => i.slug !== item.slug))"
            >
              <Icon
                name="tabler:pinned-off"
                class="size-4 min-w-4"
                aria-hidden="true"
              />
              {{ $t('components.infoCard.remove') }}
            </button>
          </div>
        </UiAccordionContent>
      </UiAccordionItem>
    </UiAccordion>
  </Card>
</template>
