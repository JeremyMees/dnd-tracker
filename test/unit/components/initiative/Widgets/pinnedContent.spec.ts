import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import PinnedContent from '~/components/initiative/Widgets/PinnedContent.vue'
import open5eItem from '~~/test/unit/fixtures/open5e-item.json'

interface Props { value: Open5eItem[] }

const props: Props = {
  value: [
    open5eItem as unknown as Open5eItem,
    open5eItem as unknown as Open5eItem,
  ],
}

describe('Initiative pinned content widget', async () => {
  it('Should match snapshot', async () => {
    const component = await mountSuspended(PinnedContent, { props })

    expect(component.html()).toMatchSnapshot()
  })

  it('Should show empty state when no items', async () => {
    const component = await mountSuspended(PinnedContent, { props: { value: [] } })

    expect(component.text()).toContain('pages.encounter.pinnedContent.empty')
  })

  it('Should show accordion when items are present', async () => {
    const component = await mountSuspended(PinnedContent, { props })

    expect(component.find('[data-test-accordion]').exists()).toBeTruthy()
  })
})
