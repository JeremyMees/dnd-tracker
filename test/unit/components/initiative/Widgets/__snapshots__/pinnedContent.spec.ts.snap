// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initiative pinned content widget > Should match snapshot 1`] = `
"<div class="transition-colors duration-300 border-4 rounded-lg p-4 border-secondary bg-secondary/50">
  <div data-test-accordion="">
    <div data-state="closed" data-orientation="vertical" class="border-b last:border-b-0">
      <h3 data-orientation="vertical" data-state="closed" class="flex"><button type="button" aria-controls="" aria-expanded="false" data-state="closed" id="reka-accordion-trigger-v-0-0-0" data-reka-collection-item="" data-orientation="vertical" class="flex flex-1 items-center justify-between py-4 font-medium transition-all [&amp;[data-state=open]>svg]:rotate-180 hover:no-underline">Aboleth<span class="iconify i-tabler:chevron-down size-4 shrink-0 transition-transform duration-200" aria-hidden="true"></span></button></h3>
      <div role="region" aria-labelledby="reka-accordion-trigger-v-0-0-0" data-state="closed" data-orientation="vertical" style="--reka-accordion-content-width: var(--reka-collapsible-content-width); --reka-accordion-content-height: var(--reka-collapsible-content-height); --reka-collapsible-content-height: 0px; --reka-collapsible-content-width: 0px;" class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down" id="reka-collapsible-content-v-0-0-1" hidden="">
        <!---->
      </div>
    </div>
    <div data-state="closed" data-orientation="vertical" class="border-b last:border-b-0">
      <h3 data-orientation="vertical" data-state="closed" class="flex"><button type="button" aria-controls="" aria-expanded="false" data-state="closed" id="reka-accordion-trigger-v-0-0-2" data-reka-collection-item="" data-orientation="vertical" class="flex flex-1 items-center justify-between py-4 font-medium transition-all [&amp;[data-state=open]>svg]:rotate-180 hover:no-underline">Aboleth<span class="iconify i-tabler:chevron-down size-4 shrink-0 transition-transform duration-200" aria-hidden="true"></span></button></h3>
      <div role="region" aria-labelledby="reka-accordion-trigger-v-0-0-2" data-state="closed" data-orientation="vertical" style="--reka-accordion-content-width: var(--reka-collapsible-content-width); --reka-accordion-content-height: var(--reka-collapsible-content-height); --reka-collapsible-content-height: 0px; --reka-collapsible-content-width: 0px;" class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down" id="reka-collapsible-content-v-0-0-3" hidden="">
        <!---->
      </div>
    </div>
  </div>
</div>"
`;
